Library/Homebrew/dev-cmd/update-python-resources.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  module_function
+
+  def update_python_resources_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `update-python-resources` [<options>] <formula>
+
+        Update versions for PyPI resource blocks in <formula>.
+      EOS
+      switch "-p", "--print-only",
+             description: "Print the updated resource blocks instead of changing the formula."
+      named :formula
+    end
+  end
+
+  def update_python_resources
+    update_python_resources_args.parse
+
+    formula = args.formulae.first
+    pypi_name = if formula.stable.url.start_with?("https://files.pythonhosted.org/packages/")
+      File.basename(formula.stable.url).match(/^(.+)-[a-z0-9.]+$/)[1]
+    else
+      formula.name
+    end
+
+    pypi_resources = formula.resources.select do |resource|
+      resource.url.start_with?("https://files.pythonhosted.org/packages/") && resource.name != "homebrew-virtualenv"
+    end
+
+    odie "No PyPI resources found in \"#{formula.name}\"" if pypi_resources.empty?
+
+    odie "\"python@3.8\" must be installed" unless Formula["python@3.8"].any_version_installed?
+
+    odie "\"pipgrip\" must be installed (`pip3 install pipgrip`)" unless File.exist?("#{HOMEBREW_PREFIX}/bin/pipgrip")
+
+    ohai "Retrieving PyPI dependencies for \"#{pypi_name}\"" unless args.print_only?
+    pipgrip_output = JSON.parse(Utils.safe_popen_read("#{HOMEBREW_PREFIX}/bin/pipgrip", "--json", pypi_name))
+
+    ohai "Updating `resource` blocks"
+    Utils::Inreplace.inreplace(formula.path) do |s|
+      pypi_resources.each do |resource|
+        new_version = pipgrip_output[resource.name]
```

Comment:
From https://github.com/Homebrew/homebrew-core/pull/58464, `resource.name` won't always exactly match the `pipgrip` output. The existing resource was `Pillow` and `pipgrip` referred to it as `pillow`, so it didn't get updated. Since names are case-insensitive, maybe lowercase both prior the existing resource name and the `pipgrip` output before trying the lookup.

Reply:
Good idea. Thanks!

Reply:
It gets even worse, see second caveat: https://github.com/ddelange/pipgrip/blob/0.5.0/README.md#known-caveats

Here is the `canonicalize_name` logic used under the hood by pip and pipgrip: https://github.com/pypa/packaging/blob/20.4/packaging/utils.py#L19